recirq/otoc/parallel_xeb.py

```diff
@@ -0,0 +1,725 @@
+
+        err_min = 1.0
+        soln_vec = np.zeros(5)
+        if gate_to_fit == 'sqrt-iswap':
+            init_guess = np.zeros(3)
+            bounds = (np.ones(3) * -1.0, np.ones(3) * 1.0)
+        elif gate_to_fit == 'iswap' or gate_to_fit == 'cz':
+            init_guess = np.zeros(2)
+            bounds = (np.ones(2) * -1.0, np.ones(2) * 1.0)
+        else:
+            init_guess = np.array([0.0, 0.0, 0.0, 0.0, 0.1])
+            bounds = (np.ones(5) * -0.6, np.ones(5) * 0.6)
+
+        for _ in range(num_restarts):
+            res = pybobyqa.solve(
+                cost_function, init_guess, maxfun=3000, bounds=bounds,
+                rhoend=1e-11)
+
+            if gate_to_fit == 'sqrt-iswap':
+                init_guess = np.random.uniform(-0.3, 0.3, 3)
+            elif gate_to_fit == 'iswap' or gate_to_fit == 'cz':
+                init_guess = np.random.uniform(-0.3, 0.3, 2)
+            else:
+                init_guess = np.random.uniform(-0.2, 0.2, 5)
+                init_guess[0] = 0.0
+                init_guess[4] = 0.1
+
+            if res.f < err_min:
+                err_min = res.f
+                if gate_to_fit == 'sqrt-iswap':
+                    soln_vec = np.zeros(5)
+                    soln_vec[1:4] = np.asarray(res.x)
+                elif gate_to_fit == 'iswap':
+                    soln_vec = np.zeros(5)
+                    soln_vec[1:3] = np.asarray(res.x)
+                elif gate_to_fit == 'cz':
+                    soln_vec = np.zeros(5)
+                    soln_vec[1] = np.asarray(res.x)[0]
+                    soln_vec[3] = np.asarray(res.x)[1]
+                else:
+                    soln_vec = np.asarray(res.x)
+
+        err_1, f_vals_1, x_fitted_1, y_fitted_1, _ = xeb_fidelity(
+            soln_vec, num_p=len(num_cycle_range))
+        final_errors_optimized[(q0, q1)] = err_1
+
+        delta_angles[(q0, q1)] = {a: soln_vec[i] for i, a in
+                                  enumerate(_fsim_angle_labels)}
+
+        new_angles = fsim_angles.copy()
+        for k, v in new_angles.items():
+            new_angles[k] += delta_angles[(q0, q1)][k]
+
+        fitted_angles[(q0, q1)] = new_angles
+
+        q_0 = cirq.GridQubit(*q0)
+        q_1 = cirq.GridQubit(*q1)
+
+        gate_list = generic_fsim_gate(new_angles, (q_0, q_1))
+        circuit_fitted = cirq.Circuit(gate_list)
+        fitted_gates[(q0, q1)] = circuit_fitted
+
+        corrected_angles = new_angles.copy()
+        corrected_angles['delta_plus'] *= -1.0
+        corrected_angles['delta_minus_off_diag'] *= -1.0
+        corrected_angles['delta_minus_diag'] *= -1.0
+        corrected_angles['theta'] = fsim_angles['theta']
+        corrected_angles['phi'] = fsim_angles['phi']
+        gate_list_corrected = generic_fsim_gate(corrected_angles, (q_0, q_1))
+        circuit_corrected = cirq.Circuit(gate_list_corrected)
+        correction_gates[(q0, q1)] = circuit_corrected
+
+        fig = plt.figure()
```

Comment:
separate plotting into its own function and have the user call it explicitly with the results of this function (if they desire)

Reply:
Done. Plotting is now a separate function call.